minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/MinecraftGameProvider.java

```diff
@@ -446,14 +447,19 @@ public void launch(ClassLoader loader) {
 			targetClass = "net.fabricmc.loader.impl.game.minecraft.applet.AppletMain";
 		}
 
+		MethodHandle invoker;
+
 		try {
 			Class<?> c = loader.loadClass(targetClass);
-			Method m = c.getMethod("main", String[].class);
-			m.invoke(null, (Object) arguments.toArray());
-		} catch (InvocationTargetException e) {
-			throw new FormattedException("Minecraft has crashed!", e.getCause());
-		} catch (ReflectiveOperationException e) {
+			invoker = MethodHandles.lookup().findStatic(c, "main", MethodType.methodType(Void.TYPE, String[].class));
```

Comment:
These should use void.class, Void.TYPE is the older / desugared form

Reply:
Changed.